zstash/ls.py

```diff
@@ -100,8 +100,14 @@ def ls_database(args: argparse.Namespace, cache: str) -> List[FilesRow]:
                 hpss = config.hpss
             else:
                 raise TypeError("Invalid config.hpss={}".format(config.hpss))
-            # Retrieve from HPSS
-            hpss_get(hpss, get_db_filename(cache), cache)
+            try:
+                # Retrieve from HPSS
+                hpss_get(hpss, get_db_filename(cache), cache)
+            except RuntimeError:
+                # Ignore failed commands
+                # If the match list is empty, then `zstash ls` will display nothing,
+                # just as `ls` does.
+                return []
```

Comment:
I don't think this is getting used. The "nothing to ls" message still gets printed in some cases.